### PR TITLE
[Instrumentation.AspNetCore] Fix changelog entry and remove package reference

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #
Design discussion issue #

## Changes

Moved changelog entry to its corresponding project and addressed unresolved item from https://github.com/open-telemetry/opentelemetry-dotnet/pull/5136#discussion_r1418016892 (Package reference is not needed to use Features in net6.0)

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 ## Unreleased
 
+* Fixed an issue where the created activity from ASP.NET Core was replaced
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:5]: the changelog entry is hard to follow. Maybe better to link to https://github.com/open-telemetry/opentelemetry-dotnet/issues/4466, and word like
Fix for "IHttpActivityFeature.Activity and Activity.Current don't return the same activity"

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md]: updated

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 ## Unreleased
 
+* Fixed an issue where the activity instance returned by `Activity.Current` was
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:5]: Fixed Issue [#4466](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4466), where the activity instance returned by `Activity.Current` was
different than instance obtained from `IHttpActivityFeature.Activity`.
[#5136](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5136)

**Review by alanwest — APPROVED** [review]
